eng/common/TestResources/New-TestResources.ps1

```diff
@@ -119,6 +119,22 @@ function MergeHashes([hashtable] $source, [psvariable] $dest) {
     }
 }
 
+function BuildBicepFile([System.IO.FileSystemInfo] $file) {
+    $tmp = $env:TEMP ? $env:TEMP : [System.IO.Path]::GetTempPath()
+    $prefix = ($file | Resolve-Path -Relative) -replace '\.|/', '_'
```

Comment:
Do you really want to resolve the full path? That might make the file prefix pretty long in some contexts.

Reply:
I was thinking about this. I think just resolving the parent directory isn't enough to guarantee uniqueness in perpetuity (or enough so that if you're debugging the compiled output, you know where it came from). The other option is to parse the path to figure out where the git root starts and chop the rest off, but that struck me as a lot more logic than necessary. So in the absence of an obvious shorter solution, I went with full path, but I'm open to ideas.

Reply:
I guess we could just use a guid or date, and then the logs from the script output can show you which file to inspect?

Reply:
If we want to avoid potential clashes then using a guid might be better, especially if we start cleaning it up at the end of this script.

Reply:
Yeah, I like that approach. I'll switch to that.